src/sentry/utils/appleconnect/appstore_connect.py

```diff
@@ -112,6 +112,48 @@ def _get_appstore_info_paged(
         next_url = _get_next_page(response)
 
 
+_RelType = NewType("_RelType", str)
+_RelId = NewType("_RelId", str)
+
+
+class _IncludedRelations:
+    """Related data which was returned with a page.
+
+    The API allows to add an ``&include=some,types`` query parameter to the URLs which will
+    automatically include related data of those types which are referred in the data of the
+    page to be returned in the same request.  This class extracts this information from the
+    page and makes it available to look up.
+
+    :param data: The entire page data, the constructor will extract the included relations
+       from this.
+    """
+
+    def __init__(self, page_data: JSONData):
+        self._items: Dict[Tuple[_RelType, _RelId], JSONData] = {}
+        for relation in page_data.get("included", []):
+            rel_type = _RelType(relation["type"])
+            rel_id = _RelId(relation["id"])
+            self._items[(rel_type, rel_id)] = relation
+
+    def get_related(self, data: JSONData, relation: str) -> Optional[JSONData]:
+        """Returns the named relation of the object.
+
+        ``data`` must be a JSON object which has a ``relationships`` object and
+        ``relation`` is the key of the specific related data in this list required.  This
+        function will read the object type and id from the relationships and look up the
+        actual object in the page's related data.
+        """
+        rel_ptr_data = safe.get_path(data, "relationships", relation, "data")
+        if rel_ptr_data is None:
+            # E.g. a query asks for both the appStoreVersion and preReleaseVersion relations
+            # to be included.  However for each build there could be only one of these that
+            # will have the data with type and id, the other will have None for data.
+            return None
+        rel_type = _RelType(rel_ptr_data["type"])
+        rel_id = _RelId(rel_ptr_data["id"])
+        return self._items[(rel_type, rel_id)]
```

Comment:
should this also be a `self._items.get((rel_type, rel_id), None)`, or would it be preferable to throw an exception immediately if this relation couldn't be found? 

this scenario could occur when `included` is an empty-ish value during `__init__`, ie the scenario the PR is handling. no items will be inserted if `included` doesn't have anything to iterate over, which gives me the impression that we'll get a `KeyError` if the issue this is attempting to fix is reproduced.

Reply:
maybe, so far this is guessing though.  I'd really like to see the error happening as that'll give me more information.  maybe i'm not defensive enough... but having sentry give me errors like this is really nice 😁

Reply:
hm, could we write unit tests for this class? i don't think it should be a blocker for this PR as we'll fail pretty quickly with a `KeyError` if the aforementioned scenario does occur, but it would be great if we had unit tests to verify and document some of the assumptions we're making here. 

something along the lines of

```py
included = IncludedRelations(json.loads(test_json))
included.get_related(...)
```

where `test_json` includes values like
`{ included: [], relationships: ... }`, `{ relationships: ... }`, `{ included: [relation], relationships: ...}`. 

data fed into `get_related` were your usual test values, e.g. `relation` is empty string, non-member string, member string and `rel_ptr_data` is a value that produces a hit in `self._items` and also a value that doesn't produce a value in `self._items`.